gulpfile.babel.js

```diff
@@ -0,0 +1,135 @@
+    'ie_mob >= 10',
+    'ff >= 30',
+    'chrome >= 34',
+    'safari >= 7',
+    'opera >= 23',
+    'ios >= 7',
+    'android >= 4.4',
+    'bb >= 10'
+  ];
+
+  // For best performance, don't add Sass partials to `gulp.src`
+  return gulp.src([
+    'static/sass/**/*.scss'
+  ])
+    .pipe($.sass({
+      outputStyle: 'compressed',
+      precision: 10
+    }).on('error', $.sass.logError))
+//    .pipe($.autoprefixer(AUTOPREFIXER_BROWSERS))
+    .pipe(gulp.dest('static/css'));
+});
+
+gulp.task('scripts', () => {
+  // Minify the vulcanized script files (overwriting in place)
+  //  - The 'base' option is used to retain the relative path of each file
+  return gulp.src(
+    'static/elements/*.vulcanize.js', {base: 'static'}
+  )
+    .pipe($.uglify())
+    .pipe(gulp.dest('static'));
+});
+
+gulp.task('vulcanize', () => {
+  // Vulcanize the Polymer imports, creating *.vulcanize.* files beside the
+  // original input files.
+  return gulp.src([
+      'static/elements/elements.html',
+      'static/elements/metrics-imports.html',
+      'static/elements/features-imports.html',
+      'static/elements/admin-imports.html',
+      'static/elements/samples-imports.html',
+    ], {base: 'static'}
+  )
+    // Vulcanize does not allow the name of the output file to be changed, so
+    // must manually rename the file first.
+    .pipe($.rename({suffix: '.vulcanize'}))
+    .pipe(gulp.dest('static'))
+    .pipe($.vulcanize({
+      stripComments: true,
+      inlineScripts: true,
+      inlineCss: true
+    }))
+    // Create an external script file to satisfy CSP
+    .pipe($.crisper({
+      scriptInHead: false
+    }))
+    .pipe(gulp.dest('static'));
+});
+
+// Clean output directory
+gulp.task('clean', () => del(['static/elements/*.vulcanize.{html,js}'], {dot: true}));
+
+// Build production files, the default task
+gulp.task('default', ['clean'], cb =>
+  runSequence(
+    'styles',
+    'vulcanize',
+    'scripts',
+//    'generate-service-worker',
+    cb
+  )
+);
+
+// Copy over the scripts that are used in importScripts as part of the generate-service-worker task.
+gulp.task('copy-sw-scripts', () => {
+  return gulp.src(['node_modules/sw-toolbox/sw-toolbox.js', 'app/scripts/sw/runtime-caching.js'])
+    .pipe(gulp.dest('dist/scripts/sw'));
+});
+
+// See http://www.html5rocks.com/en/tutorials/service-worker/introduction/ for
+// an in-depth explanation of what service workers are and why you should care.
+// Generate a service worker file that will provide offline functionality for
+// local resources. This should only be done for the 'dist' directory, to allow
+// live reload to work as expected when serving from the 'app' directory.
+gulp.task('generate-service-worker', ['copy-sw-scripts'], () => {
+  const rootDir = 'dist';
+  const filepath = path.join(rootDir, 'service-worker.js');
+
+  return swPrecache.write(filepath, {
+    // Used to avoid cache conflicts when serving on localhost.
+    cacheId: pkg.name || 'web-starter-kit',
```

Comment:
This task isn't used right now, so I left it as copied from the Web Starter Kit. I'm working on adding a service worker for #329, and there are a lot more changes there. I'd prefer to leave as is for now. Or I could delete it entirely, and add back the finished task later.